ops/testing.py

```diff
@@ -278,15 +332,15 @@ def begin_with_initial_hooks(self) -> None:
             # To be fired.
         """
         self.begin()
-
+        my_charm = cast(CharmBase, self._charm)
         # Checking if disks have been added
         # storage-attached events happen before install
         for storage_name in self._meta.storages:
             for storage_index in self._backend.storage_list(storage_name, include_detached=True):
-                s = model.Storage(storage_name, storage_index, self._backend)
+                s = model.Storage(storage_name, storage_index, self._backend)  # type: ignore
```

Comment:
Hum, interesting... I got an error before, because `self._backend: _TestingModelBackend`, while `Storage` wants `_ModelBackend`.
But indeed pyright seems not to complain (anymore?). I have a feeling that it should, however. Anywhoos, will get rid of that.